src/Roave/SecurityAdvisories/FriendsOfPhp.hs

```diff
@@ -0,0 +1,45 @@
+{-# LANGUAGE OverloadedStrings #-}
+module Roave.SecurityAdvisories.FriendsOfPhp (
+  parseAdvisoryYaml,
+  Advisory(..),
+  AffectedBranch(..)
+) where
+
+import Data.Map (Map)
+import Data.ByteString (ByteString)
+import Data.Text (Text)
+import Data.Yaml
+import qualified Data.Yaml as Y
+
+data AffectedBranch = AffectedBranch {
+  time :: Maybe Text
+  , versions :: [Text]
+} deriving (Eq, Show)
+
+data Advisory = Advisory {
+  title :: Maybe Text
+  , link :: Maybe Text
+  , cve :: Maybe Text
+  , branches :: Map Text AffectedBranch
+  , reference :: Maybe Text
+} deriving (Eq, Show)
+
+parseAdvisoryYaml :: ByteString -> Either String Advisory
+parseAdvisoryYaml = Y.decodeEither
```

Comment:
Wanted to use `decodeEither'`, but it doesn't provide an instance of `Eq`, so it cannot be used while parsing. `decodeEither` is deprecated. Any ideas welcome :+1:

Reply:
I'm getting this type error:

```
    • Couldn't match type ‘ParseException’ with ‘[Char]’
      Expected type: ByteString -> Either String Advisory
        Actual type: ByteString -> Either ParseException Advisory
    • In the expression: decodeEither'
      In an equation for ‘parseAdvisoryYaml’:
          parseAdvisoryYaml = decodeEither'
   |
29 | parseAdvisoryYaml = decodeEither'
   |                     ^^^^^^^^^^^^^
```

Reply:
Yes, that's because I added the tyoe declaration on `parseAdvisoryYaml`: if you adapt it, it is not possible to use HSpec's `shouldBe` with this type

Reply:
Gotcha, let's delay the decision until we know what kind of error we need.

Reply:
You can use the BiFunctor instance for Either to map the left side. What you'll need is a morphism from `ParseException` to `String` (probably want to use Text here).

https://hackage.haskell.org/package/bifunctors-5/docs/Data-Bifunctor.html

Reply:
If I understood this correctly, what you are suggesting is to have an `instance Bifunctor ParseException where ...` in order for the test to remain unchanged?